core/controllers/base.py

```diff
@@ -245,9 +261,11 @@ def __init__(self, request, response):  # pylint: disable=super-init-not-called
                             user_settings.last_logged_in)):
                     user_services.record_user_logged_in(self.user_id)
 
-        self.roles = (
-            [feconf.ROLE_ID_GUEST]
-            if self.user_id is None else user_settings.roles)
+            self.roles = user_settings.roles
+
+        if self.user_id is None:
+            self.roles = [feconf.ROLE_ID_GUEST]
```

Comment:
Reason this:

Reply:
I did these changes because MyPy is unable to find the definition of `user_settings`, because `user_settings` is defined inside the if clause. So according to python, we are accessing `user_settings` outside of the scope and due to this MyPy is unable to find the type of `user_settings`.